packages/core/src/components/carousel/carousel.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Prop,
+  h,
+} from '@stencil/core'
+import 'swiper/swiper-element-bundle.js'
+
+import { AutoplayOptions, PaginationOptions, Swiper } from 'swiper/types'
+
+interface SwiperElement extends HTMLElement {
+  swiper: Swiper
+}
+@Component({
+  tag: 'atom-carousel',
+  styleUrl: 'carousel.scss',
+  shadow: false,
+})
+export class AtomCarousel {
+  @Element() host!: HTMLElement
+  private swiperEl: SwiperElement
+  @Event() atomClickPrev!: EventEmitter<string>
+  @Event() atomClickNext!: EventEmitter<string>
+  @Event() atomChange!: EventEmitter<string>
+
+  @Prop({ mutable: true })
+  autoplay?: boolean
+  @Prop({ mutable: true })
+  autoplayDelay?: AutoplayOptions['delay']
+  @Prop({ mutable: true })
+  freeMode?: boolean
+  @Prop({ mutable: true })
+  loop?: boolean
+  @Prop({ mutable: true })
+  navigation?: boolean = true
+  @Prop({ mutable: true })
+  pagination?: boolean = true
+  @Prop({ mutable: true })
+  paginationClickable?: boolean
+  @Prop({ mutable: true })
+  paginationType?: PaginationOptions['type'] = 'bullets'
+  @Prop({ mutable: true })
+  slidesPerGroup?: number | string = 1
+  @Prop({ mutable: true })
+  slidesPerView?: number | string = 1
+  @Prop({ mutable: true })
+  spaceBetween?: number = 0
+  @Prop({ mutable: true }) speed?: number
+
+  // injectStyles is needed in order to overwrite css by class inside shadow root with ::part
+  private injectStyles = ['.swiper-button-disabled { opacity: 0 !important}']
+
+  private convertChildren() {
+    this.swiperEl.innerHTML = this.host.innerHTML.replace(
+      /atom-carousel-item/g,
+      'swiper-slide'
+    )
+    let idx = 0
+    while (idx < this.host.children.length) {
+      if (this.host.children[idx].outerHTML.indexOf('atom-carousel-item') > -1)
+        this.host.removeChild(this.host.children[idx])
+      else idx++
+    }
```

Comment:
Suggestion; Another option to it would be using `for` instead: 

```TS
for (const element of this.host.children) {
  if (element.outerHTML.indexOf('atom-carousel-item') > -1)
    this.host.removeChild(element)
}

// or even


for (let idx=0;idx < this.host.children.length;idx++) {
  if (this.host.children[idx].outerHTML.indexOf('atom-carousel-item') > -1)
    this.host.removeChild(this.host.children[idx])
}
```


Totally up to you, but guess it works as well

Reply:
@tassioFront i confess i havent even tried for. Several years ago i have faced big trouble with array manipulation/deletion inside of the array loop iself, like on the first example. Since then ialways go the to safe path 😅 

Besides that, In fact the while loop is a bit of different from the for, see:
On the while code i delete the children and the index keep the same, so when i delete array[2], the next array[3] becames the new array[2], but i keep deletin the array[2] because my idx havent change.
The for otherwise increments and jump to next, "forgeting" that the previously 3 now became 2, and then goes straight to the array[3] to delete it, this way the previously 3 that became the new 2  is never getting deleted.

Least but no last you may ask why i have the valitions with else, this is because the way component renders.
<atom-component> will have to kind of children <atom-component-item> the one we will remove from here and only existe so we get the html to be used in referente, and the correct <swiper-container>, whitout that if/else we would remove the swiper-container too.

Does this make sense to you?

Reply:
That makes all sense! Thanks for the clear explanation!